onnxruntime/python/tools/quantization/operators/attention.py

```diff
@@ -21,7 +21,7 @@ def quantize(self):
         assert (node.op_type == "Attention")
 
         (quantized_input_names, zero_point_names, scale_names, nodes) = \
-            self.quantizer.quantize_inputs(node, [0, 1])
+            self.quantizer.quantize_inputs(node, [0, 1], True, False, True, True)
 
```

Comment:
It should be configurable. QLinearConv does per-channel like this. And it is same thing for reduce_range. It should be configurable.
[https://github.com/microsoft/onnxruntime/blob/7e42840298aa09b35a709bd327edca110691d6eb/onnxruntime/python/tools/quantization/operators/conv.py#L80](https://github.com/microsoft/onnxruntime/blob/7e42840298aa09b35a709bd327edca110691d6eb/onnxruntime/python/tools/quantization/operators/conv.py#L80)

Reply:
For the best accuracy, it's usually better to control these options per operator. Any suggestions for this?

Reply:
e.g. For embedding, we want to utilize full 8-bit precision and per-tensor quantization, because channel dimension is big and in-efficient. On the other hand, for GEMM (attention, matmul) operations, we want to use 7-bit to avoid overflow and per-column which is critical for the accuracy.

Reply:
Some more details, For the embedding matrix, per-channel (which needs to be the 1st dim) requires a special de-quantization which de-quantize every row separately which might be inefficient. For GEMMs, we have special kernels which can handle per-channel inputs efficiently.